daemon/emer-persistent-cache.c

```diff
@@ -501,121 +431,99 @@ compute_boot_offset (EmerPersistentCache *self,
-                                              &absolute_time);
+      g_propagate_prefixed_error (error, local_error,
+                                  "Failed to read boot offset from metadata "
+                                  "file %s. ",
+                                  priv->boot_metadata_file_path);
+      return FALSE;
     }
 
   uuid_t saved_boot_id, system_boot_id;
-  if (!get_saved_boot_id (self, saved_boot_id, &error) ||
-      !get_system_boot_id (self, system_boot_id, &error))
-    {
-      g_critical ("Failed to access boot ids for comparison. Error: %s.",
-                  error->message);
-      g_error_free (error);
-      return FALSE;
-    }
+  if (!get_saved_boot_id (self, saved_boot_id, error) ||
+      !get_system_boot_id (self, system_boot_id, error))
+    return FALSE;
 
   if (uuid_compare (saved_boot_id, system_boot_id) == 0)
     {
       if (always_update_timestamps)
         {
-          gboolean write_success =
-            save_timing_metadata (self, &relative_time, &absolute_time,
-                                  NULL, NULL, NULL, &error);
-
-          if (!write_success)
-            {
-              g_warning ("Failed to update relative and absolute time in "
-                         "metadata file. Error: %s.", error->message);
-              g_error_free (error);
-            }
+          gboolean write_succeeded =
+            save_timing_metadata (self, &relative_time, &absolute_time, NULL,
+                                  NULL, NULL, error);
+          if (!write_succeeded)
+            return FALSE;
         }
-
-      return TRUE;
     }
-
-  if (!compute_boot_offset (self, relative_time, absolute_time, &boot_offset))
-    return FALSE;
-
-  gchar system_boot_id_string[BOOT_ID_FILE_LENGTH];
-  uuid_unparse_lower (system_boot_id, system_boot_id_string);
-
-  gboolean was_reset = FALSE;
-  gboolean write_success =
-    save_timing_metadata (self, &relative_time, &absolute_time, &boot_offset,
-                          system_boot_id_string, &was_reset, &error);
-
-  if (!write_success)
+  else
     {
-      g_warning ("Failed to write computed boot offset. Resetting cache. "
-                 "Error: %s.", error->message);
-      g_error_free (error);
-      return reset_boot_offset_metadata_file (self, &relative_time,
-                                              &absolute_time);
+      gboolean compute_succeeded =
+        compute_boot_offset (self, relative_time, absolute_time, &boot_offset,
+                             error);
+      if (!compute_succeeded)
+        return FALSE;
+
+      gchar system_boot_id_string[BOOT_ID_FILE_LENGTH];
+      uuid_unparse_lower (system_boot_id, system_boot_id_string);
+
+      gboolean was_reset = FALSE;
+      gboolean write_succeeded =
+        save_timing_metadata (self, &relative_time, &absolute_time,
+                              &boot_offset, system_boot_id_string, &was_reset,
+                              error);
+
+      if (!write_succeeded)
```

Comment:
It seems pretty pointless in retrospect to react to a failure to write the boot metadata file by writing the boot metadata file.